lib/archethic/self_repair/sync/transaction_handler.ex

```diff
@@ -35,18 +35,44 @@ defmodule Archethic.SelfRepair.Sync.TransactionHandler do
           transaction_summary: %TransactionSummary{
             address: address,
             type: type,
-            movements_addresses: mvt_addresses
+            movements_addresses: mvt_addresses,
+            genesis_address: genesis_address,
+            version: summary_version
           }
         },
         node_list
       ) do
-    chain_storage_nodes = Election.chain_storage_nodes_with_type(address, type, node_list)
+    last_chain_storage_nodes = Election.chain_storage_nodes_with_type(address, type, node_list)
+    genesis_chain_storage_nodes = Election.chain_storage_nodes(genesis_address, node_list)
+    node_key = Crypto.first_node_public_key()
 
-    if Utils.key_in_node_list?(chain_storage_nodes, Crypto.first_node_public_key()) do
+    if Utils.key_in_node_list?(last_chain_storage_nodes, node_key) or
+         Utils.key_in_node_list?(genesis_chain_storage_nodes, node_key) do
       not TransactionChain.transaction_exists?(address)
     else
       io_node? =
-        Enum.any?(mvt_addresses, fn address ->
+        mvt_addresses
+        |> Task.async_stream(fn address ->
+          case summary_version do
+            # Before AEIP-21, we fetch the genesis address to avoid multiple beacon chain's data changes
+            1 ->
```

Comment:
Here you assume that if the transaction summary is version 2, it means the resolved movement address are the genesis, but this is not true. During phase 1, transaction summary will be version 2, but resolved movement address will still be the last address
The only true condition to ensure the behavior of the resolved movement address is to check the protocol version which is not available in the transaction summary.

So we have 2 options, either have a condition for the resolved movement base on the transaction summary version, which mean we will need to have a transaction summary version 3 after phase 2 without any modification of the transaction summary. Or put the protocol version in the transaction summary

Reply:
Not false, then I would prefer to increase the summary's version in order to reduce storage size for the beacon, as it's not particularly necessary.